hs_core/models.py

```diff
@@ -3042,6 +3042,22 @@ def has_logical_file(self):
         """Check existence of logical file."""
         return self.logical_file is not None
 
+    @property
+    def get_or_create_logical_file(self):
+        """ create a logical file on the fly if necessary """
```

Comment:
May be better to say here that it only creates GenericLogicalFile if logical file is missing.